core/test/nuxeo-audit-page-provider.test.js

```diff
@@ -0,0 +1,212 @@
+  server.respondWith('POST', '/api/v1/automation/Audit.QueryWithPageProvider', ({ body }) => {
+    // build request parameters and sort (optional) from the body
+    const { currentPageIndex, pageSize, providerName, namedQueryParams, sortBy, sortOrder } = body.params;
+    const flatedQueryParams = {};
+    Object.entries(namedQueryParams).forEach(([key, value]) => {
+      flatedQueryParams[key] = key === 'eventIds' ? JSON.parse(value) : value;
+    });
+    let bodyParams = { currentPageIndex, pageSize, providerName, ...flatedQueryParams };
+    if (sortBy && sortOrder) {
+      bodyParams = { sortBy, sortOrder, ...bodyParams };
+    }
+
+    // build the expected parameters (add the commons properties)
+    const expectedRequestParam = { currentPageIndex: 0, pageSize: 40, providerName: 'EVENTS_VIEW', ...requestParams };
+
+    expect(expectedRequestParam).to.eql(bodyParams);
+    return { 'entity-type': 'logEntries', entries: [] };
+  });
+}
+
+function checkResponse(response) {
+  expect(response).to.be.an('object');
+  expect(response).to.have.property('entity-type');
+  expect(response).to.have.property('entries');
+  expect(response.entries).to.be.an('array').that.is.empty;
+}
+
+suite('nuxeo-audit-page-provider', () => {
+  setup(() => {
+    server = fakeServer.create();
+  });
+
+  teardown(() => {
+    server.restore();
+  });
+
+  suite('Fetch Document History', () => {
+    test('Should get history without parameters', async () => {
+      mockDocHistoryResponse();
+      const provider = await getProvider({}, documentId);
+      checkResponse(await provider.fetch());
+    });
+
+    test('Should get history by principal user', async () => {
+      mockDocHistoryResponse({ principalName: 'Administrator' });
+      const provider = await getProvider({ principalName: 'Administrator' });
+      checkResponse(await provider.fetch());
+    });
+
+    test('Should get history by events', async () => {
+      mockDocHistoryResponse({ eventId: events });
+      const provider = await getProvider({ eventIds: events }, documentId);
+      checkResponse(await provider.fetch());
+    });
+
+    test('Should get history by category', async () => {
+      mockDocHistoryResponse({ category: eventDocumentCategory });
+      const provider = await getProvider({ eventCategory: eventDocumentCategory });
+      checkResponse(await provider.fetch());
+    });
+
+    test('Should get history by date range', async () => {
+      mockDocHistoryResponse({ startEventDate: from, endEventDate: to });
+      const provider = await getProvider({ startDate: from, endDate: to });
+      checkResponse(await provider.fetch());
+    });
+
+    test('Should get history by params', async () => {
+      mockDocHistoryResponse(requestParameters);
+      const provider = await getProvider(providerParameters);
+      checkResponse(await provider.fetch());
+    });
+
+    test('Should get sorted history by params', async () => {
```

Comment:
It is cool to have all these tests, but we may be able to reduce them (and not losing coverage and confidence) by focusing on two main things, since they are the element's inner logic:
- The "dance" between calling the REST endpoint vs Automation;
- The parameters transformation (also related to the one above);

For example, we may want to have a test like (and the reverse):
```
test('Should call operation when no document identifier is provided')
```

Not sure if you agree, but I think we could end with less code and the same confidence on what we are testing 🙂

Reply:
yeah :), your ideas is to reduce the number of the test and for example keep one or two by a type (REST / Automation), if it is the case, the test can be adapted to keep only:

```
 test('Should call Rest when document identifier is provided', async () => {
      mockDocHistoryResponse();
      const provider = await getProvider(providerParameters);
      checkResponse(await provider.fetch());
      checkRequestQueryParams(requestParameters);
    });

    test('Should call Rest with sorting when document identifier is provided', async () => {
      mockDocHistoryResponse();
      const provider = await getProvider(providerParameters, documentId, providerSort);
      const qParams = {
        sortBy: 'eventDate,category,eventId,principalName',
        sortOrder: 'asc,desc,desc,asc',
        ...requestParameters,
      };
      checkResponse(await provider.fetch());
      checkRequestQueryParams(qParams);
    });
 
...  // same thing for the Operation

```

is it right ? or you see other thing

Reply:
Yes, that was my idea. 🙂 
I believe we can still assert all the other parameters (the ones we don't transform and are just forwarded, for example `pageSize`).

Reply:
One last suggestion 🙂 
When the response from the server is good, i.e. is not an error, we fire an event.
It may also be an interesting behavior to include on our tests. (we can use [waitForEvent](https://github.com/nuxeo/nuxeo-elements/blob/master/testing-helpers/test-helpers.js#L52-L69) helper)

Reply:
Thanks for your efforts on this and for adding relevant tests 👏

Reply:
you welcome, thx also for you review / time :)